packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol

```diff
@@ -101,28 +102,39 @@ contract ProtectedDataSharing is ERC721Burnable, ERC721Receiver, ManageOrders, A
      *                        Functions                                        *
      ***************************************************************************/
     function consumeProtectedData(
+        uint256 _collectionId,
         address _protectedData,
         IexecLibOrders_v5.WorkerpoolOrder calldata _workerpoolOrder,
         string calldata _contentPath
     ) external returns (bytes32) {
-        // subscription : check protectedData is avaible in Subscription & subscriber endTimestamp > block.timestamp
-        // renting : check protectedData rental for renter endTimestamp > block.timestamp
+        require(
+            (protectedDataInSubscription[_collectionId][_protectedData] &&
+                subscribers[_collectionId][msg.sender] > block.timestamp) ||
+                renters[_collectionId][msg.sender] > block.timestamp,
```

Comment:
yeah indeed it could lead to security breach. I replace with the protectedData address instead. I think we didn't need to save the collectionId here for this mapping

Reply:
Commit: e1042ef84ab4755da6094bf057f70455a3cea2d5